File: ui/store/settings/sections/appearance.ts
import { type SettingsSection } from '../types';

export type AppearanceSettings = SettingsSection;

export const initialState: AppearanceSettings = {
  id: 'appearance',
  label: 'Appearance',
  description: 'Customize the look and feel of the application',
  icon: 'LuPaintbrush',
  settings: {
    mode: {
      label: 'Mode',
      description: 'Choose between light and dark mode for the UI, or let the system decide.',
      visible: true,
      type: 'select',
      default: 'system',
      value: 'system',
      options: [
        { value: 'light', label: 'Light' },
        { value: 'dark', label: 'Dark' },
        { value: 'system', label: 'System' },
      ],
    },
    theme: {
      label: 'Theme',
      description: 'Customize the look and feel of the UI with a custom theme',
      visible: true,
      type: 'select',
      default: 'default',
      value: 'default',
      options: [
        { value: 'default', label: 'Default' },
      ],
    },
    headerSearch: {
      label: 'Header Search',
      description: 'Enable or disable the search bar in the header',
      visible: true,
      type: 'toggle',
      default: true,
      value: true,
    },
    showSettingIds: {
      label: 'Show Setting IDs',
      description: 'Show the unique identifier for each setting in the settings panel',
      visible: true,
      type: 'toggle',
      default: false,
      value: false,
    },
  },
};

File: internal/settings/appearance.go
package categories

import "github.com/omniviewdev/plugin-sdk/settings"

//nolint:gochecknoglobals // we're providing this as a package level variable
var Appearance = settings.Category{
	ID:          "appearance",
	Label:       "Appearance",
	Description: "Customize the look and feel of the application",
	Icon:        "LuPaintbrush",
	Settings: map[string]settings.Setting{
		"theme": {
			ID:          "theme",
			Type:        settings.Text,
			Label:       "Theme",
			Default:     "default",
			Description: "The color theme for the application UI",
			Options: []settings.SettingOption{
				{Value: "default", Label: "Default"},
				{Value: "solarized", Label: "Solarized"},
			},
		},
	},
}
